### PR TITLE
Move project from Gradle to Maven

### PR DESCRIPTION
**Pull request by Zero3141**
Implements changes described in [#35](https://github.com/secureCodeBox/defectdojo-client-java/issues/35#issuecomment-1663815501)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -3,90 +3,99 @@
...
 
 name: Publish Release
 
-on: workflow_dispatch
+# If input is empty we automatically bump the version
+on:
+  workflow_dispatch:
+    inputs:
+      customversion:
+        description: 'Custom version (optional)'
```
Comment (Weltraumschaf) [inline .github/workflows/release.yml:16]:
```suggestion
        description: 'Custom version e.g. 1.0.0-beta (optional)'
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -3,90 +3,99 @@
...
+        required: false
+        default: ''
+      nextversion:
+        description: 'Next version (optional)'
```
Comment (Weltraumschaf) [inline .github/workflows/release.yml:20]:
```suggestion
        description: 'Next development version e.g. 1.0.0-SNAPSHOT (optional)'
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -3,90 +3,99 @@
...
-        uses: gradle/gradle-build-action@v2.7.0
-
-      - name: Release new version
+      - name: Validate next version input # Exit when version has no -SNAPSHOT suffix
+        if: github.event.inputs.nextversion != ''
         run: |
-          git config user.name 'github-actions[bot]'
-          git config user.email 'github-actions[bot]@users.noreply.github.com'
-          ./gradlew release
+          if [[ "${{ inputs.nextversion }}" != *-SNAPSHOT ]]; then exit 1; fi
```
Comment (Weltraumschaf) [inline .github/workflows/release.yml:33]: Please comment why we exit here.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -3,90 +3,99 @@
...
-        run: ./gradlew publish
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.SCB_BOT_GPG_KEY }}
+          passphrase: ${{ secrets.SCB_BOT_GPG_PASSPHRASE }}
+          git_user_signingkey: true
+          git_tag_gpgsign: true
+          git_commit_gpgsign: true
+          git_committer_name: SecureCodeBoxBot
```
Comment (Weltraumschaf) [inline .github/workflows/release.yml:67]:
```suggestion
          git_committer_name: secureCodeBoxBot
```

File: .github/workflows/snapshot.yml [hunk 5]
```diff
@@ -3,8 +3,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # This workflow will publish a Java project with Gradle
-# For gradle wrapper validation see: https://github.com/marketplace/actions/gradle-wrapper-validation
-# For gradle build see: https://github.com/marketplace/actions/gradle-build-action
+# For maven build see: https://github.com/marketplace/actions/gradle-build-action
```
Comment (Weltraumschaf) [inline .github/workflows/snapshot.yml:6]: This links to Gradle. We do not use an action anyway, right?

**Review by Weltraumschaf — CHANGES_REQUESTED** [review]
See my comments please.

File: .github/workflows/release.yml (hunk 3 above)
Comment (Zero3141) [reply .github/workflows/release.yml:33]: This is commented three lines above

File: .github/workflows/snapshot.yml [hunk 6]
```diff
@@ -2,9 +2,8 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-# This workflow will publish a Java project with Gradle
-# For maven build see: https://github.com/marketplace/actions/gradle-build-action
-# For maven publishing see: https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-maven
+# This workflow will publish a Java project with Maven
+# See https://github.com/jonashackt/github-actions-release-maven
```
Comment (Weltraumschaf) [inline .github/workflows/snapshot.yml:6]: But we do not use this action, right?

**Review by Weltraumschaf — CHANGES_REQUESTED** [review]
There is one more 😬

**Review by Weltraumschaf — APPROVED** [review]

**Pull request merged**